File: 07. Programacion funcional/Marinoni_Macarena_TP_Programación_Funcional/TP_SistemaGestionDePedidos/src/main/java/ar/edu/tup/programacion3/entities/Producto.java
package ar.edu.tup.programacion3.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class Producto extends Base {
    private String nombre;
    private Double precio;
    private String descripcion;
    private int stock;
    private String imagen;
    private Boolean disponible;
    private Categoria categoria;
}

File: Marinoni_Macarena_TPI_Prog3/Frontend/src/types/producto.ts
export interface Product {
    id: number;
    nombre: string;
    precio: number;
    descripcion: string;
    stock: number;
    imagen: string;
    disponible: boolean;
    eliminado: boolean;
    categoriaId: number; // <- Identificador plano, como exige el F2
}